### PR TITLE
Add public form to check badges by email

### LINKED ISSUE #845: "Am I pre-registered?" form
We have a lot of attendees get confused about whether or not they're pre-registered for the correct event. As we expand out into more events, this is only going to get more confusing for people.

Finding out they're not pre-registered when they come to pick up their badge is quite upsetting for many attendees. To help prevent this problem, we can create a simple verification form anyone can use to check their registration.

Attendees would enter their email address, and uber would send them a confirmation email if that email address matches a paid record in the database. In order to prevent misuse of the form, no email would be sent if there's no record found, and uber would not send an email more than once a week (or some other arbitrary timeframe).

This form can then be blasted over social media to encourage people to make sure their preregistration went through correctly.

### PR DESCRIPTION
**Pull request by kitsuta**
Fixes #845.

### CODE REVIEW

File: uber/site_sections/preregistration.py [hunk 1]
```diff
@@ -67,6 +67,33 @@ def stats(self):
     def check_prereg(self):
         return json.dumps({'force_refresh': state.AFTER_PREREG_TAKEDOWN or state.BADGES_SOLD >= MAX_BADGE_SALES})
 
+    def check_if_preregistered(self, session, message="", **params):
+        if 'email' in params:
+            attendee = session.query(Attendee).filter(func.lower(Attendee.email) == func.lower(params['email'])).one()
+            message = "Thank you! You will receive a confirmation email if you are registered for "+EVENT_NAME_AND_YEAR+"."
+            subject = EVENT_NAME_AND_YEAR+' Registration Confirmation'
+            email_text = '''<html>
```
Comment (kitsuta) [inline uber/site_sections/preregistration.py]: When I tried to use a template with send_email, it didn't render. Unless there's a reasonably easy way to fix that, we'll unfortunately have to keep the email text inside this function for now.

Comment (EliAndrewC) [inline uber/site_sections/preregistration.py]: You probably want the ``render`` function, e.g.

```python
send_email(REGDESK_EMAIL, attendee.email, subject, render('path/to/template.html', {
    'attendee': attendee,
    # other template vars could go here
}, model=attendee)
```

Comment (kitsuta) [inline uber/site_sections/preregistration.py]: Oh, yes. That is pretty easy.

Comment (kitsuta) [inline uber/site_sections/preregistration.py]: Done!

**Comment by EliAndrewC** [comment]
Looks good.  Ship it!

**Pull request merged**